src/client/src/app/lib/components/site-list.tsx

```diff
@@ -13,40 +12,60 @@ type Props = {
 const SiteList = ({ sites }: Props) => {
   const sortedSites = sites.toSorted(sortSitesByName);
   const [filteredSites, setFilteredSites] = useState(sortedSites);
-
-  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
-    const searchInput = e.target.value.toLowerCase();
-    if (searchInput.length >= 3) {
+  const [searchValue, setSearchValue] = useState('');
+  const handleSearchClick = () => {
+    const searchQuery = searchValue.toLowerCase();
+    if (searchQuery.length >= 3) {
       setFilteredSites(
         sortedSites.filter(
           s =>
-            s.name.toLowerCase().includes(searchInput) ||
-            s.odsCode.toLowerCase() === searchInput,
+            s.name.toLowerCase().includes(searchQuery) ||
+            s.odsCode.toLowerCase() === searchQuery,
         ),
       );
     } else {
       setFilteredSites(sortedSites);
     }
   };
-
-  const debounceSearchHandler = debounce(handleInputChange, 300);
+  const handleClearClick = () => {
+    setSearchValue('');
+    setFilteredSites(sortedSites);
+  };
+  const handleSearchValueChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value);
+  };
 
   return (
     <>
-      <TextInput
-        id="site-search"
-        aria-label="site-search"
-        placeholder="Search"
-        onChange={debounceSearchHandler}
-      ></TextInput>
+      <div className="search-bar">
+        <div>
+          <TextInput
+            id="site-search"
+            label="Search active sites by name, ICB or ODS code"
+            aria-label="site-search"
```

Comment:
Aria-labels will be read out by screen readers so they should be human readable, similar to visual labels. 

The Content team should really be specifying what our aria-labels should be so you could ask them, or just make it the same as the label. I don't mind which :)